linera-service/src/cli/main.rs

```diff
@@ -606,6 +606,524 @@ impl Runnable for Job {
+                                        );
+                                    }
+                                    // Modify validators
+                                    for validator in modify_validators {
+                                        validators.insert(
+                                            validator.public_key,
+                                            ValidatorState {
+                                                network_address: validator.address,
+                                                votes: validator.votes,
+                                                account_public_key: validator.account_key,
+                                            },
+                                        );
+                                    }
+                                    // Remove validators
+                                    for public_key in remove_validators {
+                                        validators.remove(&public_key);
+                                    }
+
+                                    committee = Committee::new(validators, policy);
+                                    chain_client
+                                        .stage_new_committee(committee)
+                                        .await
+                                        .map(|outcome| outcome.map(Some))
+                                }
+                            })
+                            .await
+                            .context("Failed to stage committee")?;
+                        let Some(certificate) = maybe_certificate else {
+                            return Ok(());
+                        };
+                        info!(
+                            "Created new committee:
+{:?}",
+                            certificate
+                        );
+
+                        let time_total = time_start.elapsed();
+                        info!("Operations confirmed after {} ms", time_total.as_millis());
+                    }
+
+                    ValidatorCommand::Sync {
+                        address,
+                        mut chains,
+                    } => {
+                        let context = ClientContext::new(
+                            storage,
+                            options.context_options.clone(),
+                            wallet,
+                            signer.into_value(),
+                        );
+
+                        if chains.is_empty() {
+                            chains.push(context.default_chain());
+                        }
+
+                        let validator = context.make_node_provider().make_node(&address)?;
+
+                        for chain_id in chains {
+                            let chain = context.make_chain_client(chain_id);
+
+                            Box::pin(chain.sync_validator(validator.clone())).await?;
+                        }
+                    }
+                }
+            }
+
```

Comment:
if we start the work on splitting – can you move it to a dedicated file as well?